### PR TITLE
fix(subscription): validation rule for tag values

### PR DESCRIPTION
**Pull request by luke-taylor**
# Overview/summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

1. fixes #000

### Breaking changes

1. *Replace me*

## Testing evidence

Please provide testing evidence to show that your Pull Request works/fixes as described and documented above.

## As part of this pull request I have

- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-lz-vending/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-lz-vending/issues), for tracking and closure.
- [ ] Run and `make fmt` & `make docs` to format your code and update documentation.
- [ ] Created unit and deployment tests and provided evidence.
- [ ] Updated relevant and associated documentation.

### CODE REVIEW

File: modules/subscription/variables.tf [hunk 1]
```diff
@@ -155,9 +155,9 @@ subscription_tags = {
 DESCRIPTION
   default     = {}
   validation {
-    error_message = "Tag values must contain neither `<>%&\\?/` nor control characters, and must be between 0-256 characters."
+    error_message = "Tag values must be between 0-256 characters."
     condition = alltrue(
-      [for _, v in var.subscription_tags : can(regex("^[^<>%&\\?/[:cntrl:]]{0,256}$", v))]
+      [for _, v in var.subscription_tags : can(regex("^.{0,256}$", v))]
```
Comment (jaredfholgate) [inline modules/subscription/variables.tf:160]: The old regex excluded the special characters per the docs. Do we still care about that? https://techcommunity.microsoft.com/t5/itops-talk-blog/tags-an-easy-way-to-gain-insight-into-your-cloud-resources/ba-p/846412#:~:text=1%20Azure%20currently%20supports%20up%20to%2050%20tags,a%20maximum%20of%2050%20tag%20name%2Fvalue%20pairs.%20

Comment (luke-taylor) [reply modules/subscription/variables.tf:160]: Only Tag names don't support special chars, as it says in the doc also. Tag values only have the length limit

Comment (luke-taylor) [reply modules/subscription/variables.tf:160]: image

Comment (jaredfholgate) [reply modules/subscription/variables.tf:160]: Ah, I see. So the value is fine.

**Review by jaredfholgate — APPROVED** [review]
LGTM

**Comment by b0bu** [comment]
This has broken my implementation I use a codebase tag with link to github source as part of tagging taxonomy this is backed by a few strict policies and automation around how required tags are derived. So essentially I need `github.com/org/blahblah`. What's the background behind this? I have 10s of thousands of resources outside of the sub deployment that depend on the taxonomy as-is for automation and the like.

**Comment by luke-taylor** [comment]
> This has broken my implementation I use a codebase tag with link to github source as part of tagging taxonomy this is backed by a few strict policies and automation around how required tags are derived. So essentially I need `github.com/org/blahblah`. What's the background behind this? I have 10s of thousands of resources outside of the sub deployment that depend on the taxonomy as-is for automation and the like.

What's your scenario specifically? This change was to liberalize the validation on *tag values* so things like `/` are now allowed...

**Comment by b0bu** [comment]
I see that intention now, let me check dependabot hasn't done something dumb. Since bumping the version I'm now getting the old error for the first time...

**Review by daims99nw — COMMENTED** [review]
github.com

**Comment by b0bu** [comment]
@luke-taylor @matt-FFFFFF 
https://github.com/Azure/terraform-azurerm-lz-vending/blob/2b4d4579246ed7e650ff09b364e16643818f0b2d/modules/subscription/variables.tf#L158

tags wrong or something not merged?

Screenshot 2024-05-30 at 10 07 51

**Comment by b0bu** [comment]
was a release never created for this... ?

**Comment by luke-taylor** [comment]
> was a release never created for this... ? Last release was April 10th, this was merged April 16th.

Hey, just cut the release. v4.1.1

**Comment by b0bu** [comment]
> > was a release never created for this... ? Last release was April 10th, this was merged April 16th.
> 
> Hey, just cut the release. v4.1.1

sweet

**Comment by b0bu** [comment]
Screenshot 2024-05-30 at 14 02 18

Telemetry embedding 4.1.0 in the name, not sure what sort of impact that has on tracing accuracy.

**Pull request merged**